clients/pubsub-common/src/main/java/org/eclipse/hono/client/pubsub/publisher/PubSubPublisherFactory.java

```diff
@@ -21,6 +21,17 @@
  */
 public interface PubSubPublisherFactory {
 
+    /**
+     * Closes the publisher with the given topic if it exists.
+     * <p>
+     * This method is expected to be invoked as soon as the publisher is no longer needed.
+     *
+     * @param topic The topic of the publisher to remove.
+     * @return A future that is completed when the close operation completed or a succeeded future if no publisher
+     *         existed with the given topic.
```

Comment:
this is very confusing because a succeeded future is also completed.

Reply:
That is true, I updated the description.